### PR TITLE
New documentation using Jupyter-Notebooks. 

### LINKED ISSUE #1839: Write a proof-of-concept for notebook-based documentation
### Summary of Issue

Add a proof-of-concept for notebook-based documentation

### Issue Type

- [ ] Bug
- [ ] Enhancement  
      POEM #___ <N/A or POEM number at https://github.com/OpenMDAO/POEMs>
- [x] Docs

### Description

Write a proof-of-concept doc reproducing part of the existing OpenMDAO documentation

- http://openmdao.org/twodocs/versions/3.6.0/basic_guide/index.html#single-discipline-optimization

Use https://docs.fast.ai as an example of how to structure the layout/search bar/etc.

### Example

N/A

### Environment

N/A

Operating System: <i.e. OS X 10.14.6, Windows 10, Ubuntu 16.04>

N/A

### LINKED ISSUE #2074: Replace sphynx documentation ecosystem with the jupyter-notebook implementation from the OpenMDAO_Book repo.
### Summary of Issue

Replace sphynx documentation ecosystem with the jupyter-notebook implementation from the OpenMDAO_Book repo.

### Issue Type

- [ ] Bug
- [x ] Enhancement  
      POEM #___ <N/A or POEM number at https://github.com/OpenMDAO/POEMs>

### Example

For bugs, if at all possible, provide a short snippet which reproduces the issue.
Alternatively, link to a file in another repository where the issue is demonstrated.

For enhancements this can be left as N/A assuming the enhancement is captured in a POEM

### Environment

<For bugs, otherwise N/A>

Operating System: <i.e. OS X 10.14.6, Windows 10, Ubuntu 16.04>

Python environment: <i.e. Anaconda Python 3.7.1>

Packages: <versions of Dymos, OpenMDAO, and other pertinent Python packages. You can just paste the output of `pip freeze` or `conda list` here.>

### PR DESCRIPTION
**Pull request by Kenneth-T-Moore**
### Summary

The sphinx implementation of the OpenMDAO documentation has been replaced with jupyter-notebooks.

NOTE: Might take a few tries to get the CI build working correctly.

### Related Issues

- Resolves #2074
- Resolves #1839
- 
### Backwards incompatibilities

If you installed OpenMDAO with the [all] option on OSX, you may have version compatibility problems with the packages required for doc build.  I recommend creating a fresh Anaconda environment.

To completely build a local copy of the docs, you will also need to do the following:

1. Install the latest jupyter-notebook from the repository.  This is only temporary until their next release.

```
            git clone -q https://github.com/executablebooks/jupyter-book
            cd jupyter-book
            pip install -e .
            cd ..
```

2. Setup and launch a pyparallel server for the MPI tests in the docs.

```
            # Only need to do these once.
            jupyter serverextension enable --py ipyparallel;
            ipython profile create --parallel --profile=mpi;
            echo "c.IPClusterEngines.engine_launcher_class = 'MPIEngineSetLauncher'" >> ~/.ipython/profile_mpi/ipcluster_config.py;
            echo "c.IPClusterStart.controller_launcher_class = 'MPIControllerLauncher'" >> ~/.ipython/profile_mpi/ipcluster_config.py;

            # Need this server to be running when you build the docs. Can terminate it otherwise.
            ipcluster start -n 4 --profile=mpi &
```

### New Dependencies

New:  jupyter, nbconvert, ipyparallel, jupyter-book
Deprecated: sphinx, redbaron

### CODE REVIEW

**Review by naylor-b — APPROVED** [review]

File: .github/workflows/openmdao_test_workflow.yml [hunk 1]
```diff
@@ -191,23 +191,54 @@ jobs:
...
           coveralls --base_dir $SITE_DIR
-          
+
       - name: Make docs
         shell: bash -l {0}
         run: |
           if [[ "${{ matrix.PETSc }}" && "${{ matrix.OPTIONAL }}" == "[all]" ]]; then
+
+            cd openmdao/docs
+            if [[ "${{ secrets.SNOPT_LOCATION_71 }}" || "${{ secrets.SNOPT_LOCATION_77 }}" ]]; then
```
Comment (swryan) [inline .github/workflows/openmdao_test_workflow.yml:201]: `SNOPT_LOCATION_71` should be `SNOPT_LOCATION_72`

File: openmdao/docs/build_jupyter_book.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import fnmatch
+import shutil
+
+
+_this_file = pathlib.Path(__file__).resolve()
+REPO_ROOT = _this_file.parent
+BOOK_DIR = pathlib.Path(REPO_ROOT, 'openmdao_book')
+
+
+def copy_build_artifacts(book_dir=BOOK_DIR):
```
Comment (swryan) [inline openmdao/docs/build_jupyter_book.py:14]: it looks like this function is duplicated in a  standalone file

File: .github/workflows/openmdao_test_workflow.yml (hunk 1 above)
Comment (Kenneth-T-Moore) [reply .github/workflows/openmdao_test_workflow.yml]: Fixed

File: openmdao/docs/build_jupyter_book.py (hunk 2 above)
Comment (Kenneth-T-Moore) [reply openmdao/docs/build_jupyter_book.py]: Fixed

**Review by robfalck — APPROVED** [review]

**Pull request merged**